src/lapack/red/bidiagut/front/flamec/FLASH_Bidiag_UT_form_U_ext.c

```diff
@@ -0,0 +1,98 @@
+/*
+
+    Copyright (C) 2014, The University of Texas at Austin
+
+    This file is part of libflame and is available under the 3-Clause
+    BSD license, which can be found in the LICENSE file at the top-level
+    directory, or at http://opensource.org/licenses/BSD-3-Clause
+
+*/
+
+#include "FLAME.h"
+
+FLA_Error FLASH_Bidiag_UT_form_U_ext( FLA_Uplo  uplo,  FLA_Obj A, FLA_Obj T,
+                                    FLA_Trans transu, FLA_Obj U, dim_t depth, dim_t* blocksize )
+{
+    if ( transu == FLA_NO_TRANSPOSE ||
+         transu == FLA_CONJ_NO_TRANSPOSE )
+    {
+        if ( uplo == FLA_UPPER_TRIANGULAR )
+        {
+            FLA_QR_UT_form_Q( A, T, U );
+        }
+        else // if ( uplo == FLA_LOWER_TRIANGULAR )
+        {
+            FLA_Obj ATL, ATR,
+                    ABL, ABR;
+
+            FLA_Obj UTL, UTR,
+                    UBL, UBR;
+
+            FLA_Obj TL,  TR;
+
+            dim_t   b = ( FLA_Obj_length( A ) - 1 );
+
+            FLA_Part_1x2( T,    &TL,  &TR,     1, FLA_RIGHT );
+            FLA_Part_2x2( U,    &UTL, &UTR,
+                                &UBL, &UBR,    1, 1, FLA_TL );
+
+            if ( FLA_Obj_is( A, U ) == FALSE )
+            {
+                FLASH_Set( FLA_ONE,  UTL );
+                FLASH_Set( FLA_ZERO, UTR );
+                FLASH_Set( FLA_ZERO, UBL );
+
+                FLA_Part_2x2( A,    &ATL, &ATR,
+                                    &ABL, &ABR,    1, b, FLA_TL );
+
+                FLA_QR_UT_form_Q( ABL, TL, UBR );
+            }
+            else
+            {
+                FLA_Obj p, pt, pb;
+                FLA_Part_2x2( A,    &ATL, &ATR,
+                                    &ABL, &ABR,    1, b+1, FLA_TL );
+
+                FLASH_Obj_create( FLA_INT, b+1, 1, depth, &blocksize, &p );
+                FLA_Part_2x1( p,    &pt,
+                                    &pb, 1, FLA_BOTTOM );
+                FLASH_Set( FLA_ONE, pt );
+                FLASH_Set( FLA_ZERO, pb );
+                FLA_Apply_pivots ( FLA_RIGHT, FLA_NO_TRANSPOSE, p, ABL );
+                //FLASH_Apply_pivots ( FLA_RIGHT, FLA_NO_TRANSPOSE, p, ABL );
```

Comment:
This resulted in a "Function or conditional branch/case not yet implemented" error. This should be changed once it is